module/Company/src/Company/Service/Company.php

```diff
@@ -264,6 +348,26 @@ public function saveCategory($data = null)
         }
         $this->getCategoryMapper()->save();
     }
+
+    /**
+     * Checks if the data is valid (if nonnull), and if it is saves the label
+     *
+     * @param array $data The data to validate, and apply to the label
+     */
+    public function saveLabel($data = null)
+    {
+        if ($data != null) {
+            $labelForm = $this->getLabelForm();
+            $labelForm->setData($data);
+            if ($labelForm->isValid()) {
+                $this->saveLabel();
+                return true;
+            }
+            return;
+        }
+        $this->getLabelMapper()->save();
```

Comment:
Always use `!==` or `===` when comparing non-object types.

Does the `save` method need to be called when there was no data to be inserted?

If it still needs to be called, you could do something like:

```
if ($data !== null) {
    $this->getLabelMapper()->save();
    return;
}

// other things
```

Reply:
It does not appear to be neccesary to me, so I removed it, also for the existing code